### PR TITLE
Various CSS improvements

### LINKED ISSUE #242: bs_theme_preview() results in many warnings when a colour is modified
On Windows. Using both the CRAN version 0.2.4 and the current github version, when I run `bs_theme_preview()` and change the background colour to any colour, I get all these messages in my console:

```
--------------------
bs_theme_update(theme, bg = "#1A29B6", fg = "#000")
WARNING: Found no color leading to 3:1 contrast ratio against #1928af...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 1126:37 of ../lib/bs/scss/_variables.scss
         from line 26:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1928af...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 1136:37 of ../lib/bs/scss/_variables.scss
         from line 26:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 7:10 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050927...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 13:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `@content`
         from line 13:13 of ../lib/bs/scss/mixins/_hover.scss, in mixin `hover`
         from line 12:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050927...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 20:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 27:51 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 34:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #04061c...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 46:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 58:53 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 7:10 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050927...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 13:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `@content`
         from line 13:13 of ../lib/bs/scss/mixins/_hover.scss, in mixin `hover`
         from line 12:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050927...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 20:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 27:51 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 34:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #04061c...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 46:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 58:53 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1725A4...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 7:10 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #121d82...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 13:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `@content`
         from line 13:13 of ../lib/bs/scss/mixins/_hover.scss, in mixin `hover`
         from line 12:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #121d82...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 20:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1725A4...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 27:51 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1725A4...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 34:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #111b77...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 46:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1725A4...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 58:53 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050824...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 7:10 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #000102...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 13:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `@content`
         from line 13:13 of ../lib/bs/scss/mixins/_hover.scss, in mixin `hover`
         from line 12:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #000102...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 20:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050824...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 27:51 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050824...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 34:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against black...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 46:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050824...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 58:53 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-variant`
         from line 75:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 64:53 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-outline-variant`
         from line 81:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 88:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-outline-variant`
         from line 81:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 64:53 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-outline-variant`
         from line 81:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 88:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-outline-variant`
         from line 81:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1725A4...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 64:53 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-outline-variant`
         from line 81:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1725A4...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 88:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-outline-variant`
         from line 81:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050824...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 64:53 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-outline-variant`
         from line 81:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050824...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 88:12 of ../lib/bs/scss/mixins/_buttons.scss, in mixin `button-outline-variant`
         from line 81:14 of ../lib/bs/scss/_buttons.scss
         from line 37:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 2:10 of ../lib/bs/scss/mixins/_badge.scss, in mixin `badge-variant`
         from line 52:14 of ../lib/bs/scss/_badge.scss
         from line 48:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 7:14 of ../lib/bs/scss/mixins/_badge.scss, in mixin `@content`
         from line 19:5 of ../lib/bs/scss/mixins/_hover.scss, in mixin `hover-focus`
         from line 6:14 of ../lib/bs/scss/mixins/_badge.scss, in mixin `badge-variant`
         from line 52:14 of ../lib/bs/scss/_badge.scss
         from line 48:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 2:10 of ../lib/bs/scss/mixins/_badge.scss, in mixin `badge-variant`
         from line 52:14 of ../lib/bs/scss/_badge.scss
         from line 48:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 7:14 of ../lib/bs/scss/mixins/_badge.scss, in mixin `@content`
         from line 19:5 of ../lib/bs/scss/mixins/_hover.scss, in mixin `hover-focus`
         from line 6:14 of ../lib/bs/scss/mixins/_badge.scss, in mixin `badge-variant`
         from line 52:14 of ../lib/bs/scss/_badge.scss
         from line 48:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1725A4...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 2:10 of ../lib/bs/scss/mixins/_badge.scss, in mixin `badge-variant`
         from line 52:14 of ../lib/bs/scss/_badge.scss
         from line 48:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1725A4...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 7:14 of ../lib/bs/scss/mixins/_badge.scss, in mixin `@content`
         from line 19:5 of ../lib/bs/scss/mixins/_hover.scss, in mixin `hover-focus`
         from line 6:14 of ../lib/bs/scss/mixins/_badge.scss, in mixin `badge-variant`
         from line 52:14 of ../lib/bs/scss/_badge.scss
         from line 48:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050824...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 2:10 of ../lib/bs/scss/mixins/_badge.scss, in mixin `badge-variant`
         from line 52:14 of ../lib/bs/scss/_badge.scss
         from line 48:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050824...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 7:14 of ../lib/bs/scss/mixins/_badge.scss, in mixin `@content`
         from line 19:5 of ../lib/bs/scss/mixins/_hover.scss, in mixin `hover-focus`
         from line 6:14 of ../lib/bs/scss/mixins/_badge.scss, in mixin `badge-variant`
         from line 52:14 of ../lib/bs/scss/_badge.scss
         from line 48:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 8:12 of ../lib/bs/scss/mixins/_background-variant.scss, in mixin `bg-variant`
         from line 4:12 of ../lib/bs/scss/utilities/_background.scss
         from line 2:9 of ../lib/bs/scss/_utilities.scss
         from line 61:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 8:12 of ../lib/bs/scss/mixins/_background-variant.scss, in mixin `bg-variant`
         from line 4:12 of ../lib/bs/scss/utilities/_background.scss
         from line 2:9 of ../lib/bs/scss/_utilities.scss
         from line 61:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1725A4...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 8:12 of ../lib/bs/scss/mixins/_background-variant.scss, in mixin `bg-variant`
         from line 4:12 of ../lib/bs/scss/utilities/_background.scss
         from line 2:9 of ../lib/bs/scss/_utilities.scss
         from line 61:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050824...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 8:12 of ../lib/bs/scss/mixins/_background-variant.scss, in mixin `bg-variant`
         from line 4:12 of ../lib/bs/scss/utilities/_background.scss
         from line 2:9 of ../lib/bs/scss/_utilities.scss
         from line 61:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1A29B6...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 15:10 of ../lib/bs/scss/utilities/_background.scss
         from line 2:9 of ../lib/bs/scss/_utilities.scss
         from line 61:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 4:12 of ../bs3compat/_progress_compat.scss
         from line 6:9 of ../bs3compat/_rules.scss
         from line 66:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0A1049...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 4:12 of ../bs3compat/_progress_compat.scss
         from line 6:9 of ../bs3compat/_rules.scss
         from line 66:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1725A4...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 4:12 of ../bs3compat/_progress_compat.scss
         from line 6:9 of ../bs3compat/_rules.scss
         from line 66:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #050824...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 4:12 of ../bs3compat/_progress_compat.scss
         from line 6:9 of ../bs3compat/_rules.scss
         from line 66:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1928af...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 1126:37 of ../lib/bs/scss/_variables.scss
         from line 26:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1928af...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 1136:37 of ../lib/bs/scss/_variables.scss
         from line 26:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1825a6...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 96:12 of ../../shiny/www/shared/datepicker/scss/datepicker3.scss
         from line 77:9 of ../../shiny/www/shared/datepicker/scss/build3.scss
         from line 29:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1941b6...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 34:11 of ../../shiny/www/shared/datepicker/scss/build3.scss, in mixin `button-variant`
         from line 108:13 of ../../shiny/www/shared/datepicker/scss/datepicker3.scss
         from line 77:9 of ../../shiny/www/shared/datepicker/scss/build3.scss
         from line 29:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1825a6...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 34:11 of ../../shiny/www/shared/datepicker/scss/build3.scss, in mixin `button-variant`
         from line 137:13 of ../../shiny/www/shared/datepicker/scss/datepicker3.scss
         from line 77:9 of ../../shiny/www/shared/datepicker/scss/build3.scss
         from line 29:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1933ae...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 34:11 of ../../shiny/www/shared/datepicker/scss/build3.scss, in mixin `button-variant`
         from line 152:13 of ../../shiny/www/shared/datepicker/scss/datepicker3.scss
         from line 77:9 of ../../shiny/www/shared/datepicker/scss/build3.scss
         from line 29:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #0e155f...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 34:11 of ../../shiny/www/shared/datepicker/scss/build3.scss, in mixin `button-variant`
         from line 176:13 of ../../shiny/www/shared/datepicker/scss/datepicker3.scss
         from line 77:9 of ../../shiny/www/shared/datepicker/scss/build3.scss
         from line 29:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1825a6...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 195:13 of ../../shiny/www/shared/datepicker/scss/datepicker3.scss
         from line 77:9 of ../../shiny/www/shared/datepicker/scss/build3.scss
         from line 29:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1825a6...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 228:12 of ../../shiny/www/shared/datepicker/scss/datepicker3.scss
         from line 77:9 of ../../shiny/www/shared/datepicker/scss/build3.scss
         from line 29:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1928af...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 1126:37 of ../lib/bs/scss/_variables.scss
         from line 26:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1928af...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 1136:37 of ../lib/bs/scss/_variables.scss
         from line 26:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1928af...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 1126:37 of ../lib/bs/scss/_variables.scss
         from line 26:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1928af...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 1136:37 of ../lib/bs/scss/_variables.scss
         from line 26:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1928af...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 1126:37 of ../lib/bs/scss/_variables.scss
         from line 26:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1928af...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 1136:37 of ../lib/bs/scss/_variables.scss
         from line 26:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1928af...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 1126:37 of ../lib/bs/scss/_variables.scss
         from line 26:9 of stdin

WARNING: Found no color leading to 3:1 contrast ratio against #1928af...
         on line 39:3 of ../sass-utils/color-contrast.scss, in function `color-contrast`
         from line 1136:37 of ../lib/bs/scss/_variables.scss
         from line 26:9 of stdin
```

### PR DESCRIPTION
**Pull request by cpsievert**
Closes #242, #187, #215, and #250. Also improves `bootswatch = "minty"`'s defaults to have higher levels of color contrasting.

### CODE REVIEW

File: inst/lib/bsw/dist/minty/_bootswatch.scss [hunk 1]
```diff
@@ -25,57 +25,6 @@ $web-font-path: "https://fonts.googleapis.com/css2?family=Montserrat:wght@400;50
...
-  }
-
-  &-outline-primary {
-    color: $primary;
-  }
-
-  &-outline-secondary {
-    color: $secondary;
-  }
-
```
Comment (schloerke) [inline inst/lib/bsw/dist/minty/_bootswatch.scss:55]: We need to keep the outline classes in. Ex: `btn btn-outline-primary` in https://bootswatch.com/minty/

File: inst/lib/bsw/dist/minty/_bootswatch.scss [hunk 2]
```diff
@@ -25,57 +25,6 @@ $web-font-path: "https://fonts.googleapis.com/css2?family=Montserrat:wght@400;50
 
 .btn {
   font-family: $headings-font-family;
-
-  &,
-  &:hover {
-    color: $white;
```
Comment (schloerke) [inline inst/lib/bsw/dist/minty/_bootswatch.scss:31]: Maybe `color-contrast()`?

File: inst/lib/bsw/dist/minty/_variables.scss [hunk 3]
```diff
@@ -40,7 +40,8 @@ $dark:          $gray-800 !default;
 
 // Body
 
-$body-color:                $gray-600 !default;
+$body-color:                $gray-700 !default;
+$color-contrast-dark:      $body-color !default;
```
Comment (schloerke) [inline inst/lib/bsw/dist/minty/_variables.scss:44]: Add comment for the addition of this variable

File: tools/yarn_install.R [hunk 4]
```diff
@@ -311,6 +311,46 @@ for (patch in list.files(patch_dir, full.names = TRUE)) {
...
+    "red" = "#d33682 !default",
+    "orange" = "#fd7e14 !default",
+    "yellow" = "#cb4b16 !default",
+    "green" = "#2aa198 !default",
+    "teal" = "#20c997 !default",
+    "cyan" = "#268bd2 !default",
+    "enable-gradients" = "true !default",
+    "secondary" = "$gray-800 !default",
+    "color-contrast-dark" = "#031014 !default",
+    "color-contrast-light" = "#BBD0D0 !default"
```
Comment (schloerke) [inline tools/yarn_install.R:342]:
```suggestion
    # Darker green and lighter green than `$black` and `$white`
    "color-contrast-dark" = "#031014 !default",
    "color-contrast-light" = "#BBD0D0 !default"
```

**Review by schloerke — APPROVED** [review]
LGTM given second pass at Minty outline rules

File: inst/lib/bsw/dist/minty/_bootswatch.scss (hunk 1 above)
Comment (cpsievert) [reply inst/lib/bsw/dist/minty/_bootswatch.scss:55]: Turns out these rules are actually redundant -- the weirdness I was seeing it actually a result of https://github.com/rstudio/bslib/issues/250 (I'll fix that here as well)

**Pull request merged**